### PR TITLE
enhancement : add docs for appointment scheduling

### PR DESCRIPTION
**Pull request by 0Pranav**

### CODE REVIEW

File: erpnext_com/www/docs/user/manual/en/CRM/appointment.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+metatags:
+ description: Appointment Schedulling for CRM in ERPNext
+ keywords: Appointment Schedulling , CRM, frappe, erpnext new features, erp, open source erp, free erp, security
+---
+```
+
+**An appointment is a prearranged meeting between a lead and an employee.**
+
+Appointment document type can be used to schedule and manage interaction with a lead or opportunity.
```
Comment (akshitakl) [inline erpnext_com/www/docs/user/manual/en/CRM/appointment.md]: Hi, the definition looks very generic and not ERPNext specific, you can maybe merge both the first and the second statements and see if it works...check for grammatical errors, minor changes of a missed comma here and there are required. Do let me know if you need any help.

File: erpnext_com/www/docs/user/manual/en/CRM/appointment.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
+# Appointment
```
Comment (prasadarr) [inline erpnext_com/www/docs/user/manual/en/CRM/appointment.md]: please crop the top part in images to remove the URL bar. In the third image retake, the developer field is visible, also ensure to take screenshot without interference in the top. 

List the files in index.md.

File: erpnext_com/www/docs/user/manual/en/CRM/appointment.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+**An appointment is a prearranged meeting between a Lead and an Employee of your Company.**
+
+Appointment document type can be used to schedule and manage interaction with a [Lead](/docs/user/manual/en/CRM/lead) or an [Opportunity](/docs/user/manual/en/CRM/opportunity). 
+
+To access Appointment list, go to:
+> Home > Sales Pipeline > CRM > Appointment 
+
+## 1. Prerequisites
+
+1. Appointment Booking Settings
```
Comment (prasadarr) [inline erpnext_com/www/docs/user/manual/en/CRM/appointment.md]: link these to docs

File: erpnext_com/www/docs/user/manual/en/CRM/appointment_booking_settings.md [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+### 3.3 Number of days appointment can be books in advance
+
+This is the number of days the appointment can be booked in advance. If the Holiday List provided above ends before the date calculated using this number, appointment schedulling will be stopped at the end of holiday list end.
+
+
+## 4. Success settings
+
+### 4.1 Success Redirect URL
+
+This is the URL where the user will be redirect on creation of successful appoitnment creation via Web Portal. Does not apply to desk.
```
Comment (prasadarr) [inline erpnext_com/www/docs/user/manual/en/CRM/appointment_booking_settings.md]: what does this mean? Does not apply to desk?

File: erpnext_com/www/docs/user/manual/en/CRM/appointment.md (hunk 3 above)
Comment (0Pranav) [reply erpnext_com/www/docs/user/manual/en/CRM/appointment.md]: Added links to the docs.

File: erpnext_com/www/docs/user/manual/en/CRM/appointment_booking_settings.md (hunk 4 above)
Comment (0Pranav) [reply erpnext_com/www/docs/user/manual/en/CRM/appointment_booking_settings.md]: Rephrased to make it clearer.

File: erpnext_com/www/docs/user/manual/en/CRM/appointment_booking_settings.md [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+### 3.3 Number of days appointment can be books in advance
+
+This is the number of days the appointment can be booked in advance. If the Holiday List provided above ends before the date calculated using this number, appointment schedulling will be stopped at the end of holiday list end.
+
+
+## 4. Success settings
+
+### 4.1 Success Redirect URL
+
+This is the URL where the user will be redirect on creation of successful appoitnment creation via Web Portal. This redirect will not occur when creating appointments from within Desk UI.
```
Comment (prasadarr) [inline erpnext_com/www/docs/user/manual/en/CRM/appointment_booking_settings.md]:
```suggestion
This is the URL where the user will be redirected on creation of successful appointment creation via Web Portal. This redirect will not occur when creating appointments from within Desk UI as a [desk user](/docs/user/manual/en/setting-up/articles/difference-between-system-user-and-website-user).
```

File: erpnext_com/www/docs/user/manual/en/CRM/appointment-booking-settings.md [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+
+### 2.1  Availability of Slots
+
+Here you can set the timing in which your agents are available to attend an appointment. This is set per day of the week. Each row represents a continuous block of time, you can have multiple entries for each day of the week.
+
+For example if your agents work Monday to Friday, 9am to 5pm but with a lunch break at 1.30 for half and hour. You will need to create two entries for each day. One from 9am to 1.30pm and another from 2pm to 5pm.
+
+### 2.2 Agents
+
+This is the list of agents which will be autoassigned to appointments. The number of appointments which can exist in one timeslot also depends upon the number employees in this list.
```
Comment (prasadarr) [inline erpnext_com/www/docs/user/manual/en/CRM/appointment-booking-settings.md]:
```suggestion
This is the list of agents which will be **auto-assigned** to appointments. The number of appointments which can exist in one timeslot also depends upon the number employees in this list.
```

File: erpnext_com/www/docs/user/manual/en/CRM/appointment-booking-settings.md [hunk 7]
```diff
@@ -0,0 +1,47 @@
+# Appointment Booking Settings
+
```
Comment (prasadarr) [inline erpnext_com/www/docs/user/manual/en/CRM/appointment-booking-settings.md]: add a screenshot of this settings page

File: erpnext_com/www/docs/user/manual/en/CRM/appointment.md [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+![Appointment Details](/docs/assets/img/crm/appointment-details.png)
+
+It'll match the customer email with leads in the system and if one is found, it is linked with the document.
+If no lead is found, the appointment is marked as "Unverified" and an email is sent to the customer to confirm their email
+
+## 3. Features
+
+### 3.1 Autoassign
+
+Appointments are automatically assigned to employees as per the settings in [Appointment Booking Settings](docs/user/manual/en/CRM/appointment_booking_settings).
```
Comment (prasadarr) [inline erpnext_com/www/docs/user/manual/en/CRM/appointment.md]: which part are you referring to exactly?

File: erpnext_com/www/docs/user/manual/en/CRM/appointment.md [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+![Appointment Details](/docs/assets/img/crm/appointment-details.png)
+
+It'll match the customer email with leads in the system and if one is found, it is linked with the document.
+If no lead is found, the appointment is marked as "Unverified" and an email is sent to the customer to confirm their email
+
+## 3. Features
+
+### 3.1 Autoassign
+
+Appointments are automatically assigned to employees as per the `Agents` list in [Appointment Booking Settings](docs/user/manual/en/CRM/appointment_booking_settings). The agent with the least assignment for the day of the appointment and who is free in the scheduled time is assigned to the appointment.
```
Comment (prasadarr) [inline erpnext_com/www/docs/user/manual/en/CRM/appointment.md]: least assignment in terms of number of assignments or number of hours in assignments?
eg: there are be 1 appointment of 4 hours, 4 appointment of 1 hour each. What will be considered more?

**Pull request merged**